### PR TITLE
Add documentation to the Function type

### PR DESCRIPTION
**Pull request by snowfoxsh**
There is currently no documentation on the Function type. This gives a starting point for adding documentation.
> Completed on a plane

### CODE REVIEW

File: rnum/src/function/mod.rs [hunk 1]
```diff
@@ -2,48 +2,194 @@ use std::sync::Arc;
...
     Sub {
         lhs: Arc<Function<'f>>,
         rhs: Arc<Function<'f>>,
     },
+    /// [Multiplication](https://en.wikipedia.org/wiki/Multiplication)
     Mul {
         lhs: Arc<Function<'f>>,
         rhs: Arc<Function<'f>>,
     },
+    /// [Division](https://en.wikipedia.org/wiki/Division)
```
Comment (Vio-Eli) [inline rnum/src/function/mod.rs]: Division leads to a few articles. I'd recommend either using https://en.wikipedia.org/wiki/Division_(mathematics) or https://en.wikipedia.org/wiki/Division_algorithm, probably the first

Comment (snowfoxsh) [reply rnum/src/function/mod.rs]: I will change that!

**Comment by snowfoxsh** [comment]
Everything looks good to go with this PR. Good to merge?

**Review by Vio-Eli — APPROVED** [review]
Looks good to me!

**Pull request merged**